DataStructure/src/main/list/ResizableArray.java

```diff
@@ -112,9 +136,11 @@ public boolean remove(int num){
      * @return the index of the given element. if the element is not found then it returns -1.
      */
     public int indexOf(int num){
-        for (int j = 0; j < writeIndex; j++) {
-            if(num == list[j]){
-                return j;
+        if (!isEmpty()) {
```

Comment:
Why do you need !isEmpty() ? Duplicate code. For loop will work fine if the list is empty.